### PR TITLE
Align xd84 and xd96 with moon

### PR DESCRIPTION
**Pull request by zvecr**
## Description

#5976 highlighted some issues with the xd84/xd96 port. While updating the files, I have taken the opportunity to fix a few other things such as compatibility with `LINK_TIME_OPTIMIZATION_ENABLE = yes` and some some spelling mistakes.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xd84/rules.mk [hunk 1]
```diff
@@ -78,11 +78,10 @@ BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
...
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
 HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+LINK_TIME_OPTIMIZATION_ENABLE = yes
 
 # custom matrix setup
 CUSTOM_MATRIX = yes
-SRC = i2c_master.c custom_matrix_helper.c pca9555.c matrix.c
-
-EXTRAFLAGS += -flto
+SRC += i2c_master.c custom_matrix_helper.c pca9555.c matrix.c
```
Comment (drashna) [inline keyboards/xd84/rules.mk]: Just a heads up, i2c and other hardware dependent things should be added to `QUANTUM_LIB_SRC`, (especially) if LTO is being used.  
```suggestion
SRC += custom_matrix_helper.c pca9555.c matrix.c
QUANTUM_LIB_SRC += i2c_master.c
```

File: keyboards/xd96/rules.mk [hunk 2]
```diff
@@ -78,9 +78,8 @@ BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
...
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
 HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+LINK_TIME_OPTIMIZATION_ENABLE = yes
 
 # custom matrix setup
 CUSTOM_MATRIX = yes
-SRC = i2c_master.c custom_matrix_helper.c pca9555.c matrix.c
-
-EXTRAFLAGS += -flto
+SRC += i2c_master.c custom_matrix_helper.c pca9555.c matrix.c
```
Comment (drashna) [inline keyboards/xd96/rules.mk]:
```suggestion
SRC += custom_matrix_helper.c pca9555.c matrix.c
QUANTUM_LIB_SRC += i2c_master.c
```

**Comment by zvecr** [comment]
@drashna nice catch on the i2c_master LTO stuff, completely slipped my mind.

**Comment by drashna** [comment]
Welcome! And yeah, I've been trying to keep an eye for that stuff, since it can produce wonky effects.

**Pull request merged**